mantidproject/mantid
# ISIS SANS improvements to batch processing and beam centre finder

Fixes #16863 

The improvements address a list of items that Rob mentioned:
1.  output names in batch mode with illegal characters will have them renamed. 
2.  If a workspace does not contain a BENCH_ROT entry, then the process will fail early on
3.  The original reporting for the beam centre finder for LARMOR  was fixed

**To test:**

**Test missing Bench_Rot entry**

The test data can be found here: \olympic\Babylon5\Scratch\Anton\Testing\ISIS_Improvement\MissingBenchRot
- Add the sample data path to the data search directories.
- Start up the ISIS SANS GUI
- Set the instrument to LARMOR
- Set the user file to USER_Pappas_RB1569004_162B_ISISStat_r9818.txt
- Set the file LARMOR00010351 as the sample scatter input
- Press load
  - Confirm that an exception is thrown
  - Confirm that the error message in the log terminal mentions a missing Bench_Rot entry

**Test bad output name in Batch Process**

The test data can be found here: \olympic\Babylon5\Scratch\Anton\Testing\ISIS_Improvement\BadBatchName
- Add the sample data path to the data search directories.
- Start up the ISIS SANS GUI
- Set the instrument to SANS2DTUBES
- Select Batch Mode on the Run Numbers tab
- In the CSV file browser widget, select SANS2DTUBES_BadName.csv. This contains the output name zero_f:ee_out which has the illegal colon
- Run 1D Reduce
  - Confirm that Mantid does not crash
  - Once the reduction is done, check your output directory for a new file called zero_fee_out, ie the invalid character was removed

**Test for beam centre finder for LARMOR**

The test data can be found here: \olympic\Babylon5\Scratch\Anton\Testing\ISIS_Improvement\BeamCentreFinder
- Add the sample data path to the data search directories.
- Set the instrument to LARMOR
- Set the user file to USER_Pappas_RB1569004_162B_ISISStat_r9818.txt
- Set the file LARMOR00009839 as the sample scatter input and press load
- Switch to the Geometry tab
- Set Max.iterations to 0
- Set Current-Beta-Rear to 1
- Set Current-Y-Rear to 12
- Go to your log window and set the log level to Notice
- Press Run
  - Confirm that the output on the Geometry tab output ouputs 1 and 12 as the starting point
- Do the same thing again for Current-Beta-Rear = -1 and Current-Y-Rear = 12
  - Confirm that the output on the Geometry tab output ouputs -1 and -12 as the starting point
## Release Notes

See [here](https://github.com/mantidproject/mantid/pull/16875/commits/645a94c2a0705da3ef482e5ddf0493c937443a83)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
